insights/parsers/smartctl.py

```diff
@@ -35,12 +35,13 @@ class SMARTctl(Parser):
 
     def __init__(self, context):
         super(SMARTctl, self).__init__(context)
+        self.data = {}
         filename_re = re.compile(r'smartctl_-a_\.dev\.(?P<device>\w+)$')
         match = filename_re.search(context.path)
-        self.data = {}
         if match:
             self.device = match.group('device')
-        # Else warn?  fail?
+        else:
+            self.device = ''
```

Comment:
Good points!  I'm a bit cautious about bailing out with an exception if it doesn't match the device name, because if the CommandSpec format changes or we add extra FileSpecs or something then that could cause more things to break.

However, I think Jesse's hit the nail on the head - it's useless information if you don't know which device its attached to.  So let's go with throwing a ParseException.